### PR TITLE
Fix bug that game crush loading level3

### PR DESCRIPTION
**Pull request by silent-lindsay**
Through a global dict `foes_by_missions ` initialized and altered in new-developed file `global_foes.py`, conflict caused by redefintion of `foes_by_missions` in `load_from_tmx_manager.py` and `load_from_xml_manager.py` is solved successfully.

### CODE REVIEW

File: saves/options.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <options>
-    <language>en</language>
+    <language>zh_cn</language>
```
Comment (Grimmys) [inline saves/options.xml:2]: This shouldn't have been change.

Default language should stay to English, could you revert this change please?

File: src/services/load_from_xml_manager.py [hunk 2]
```diff
@@ -38,7 +38,8 @@
 RACES_DATA_PATH = "data/races.xml"
 CLASSES_DATA_PATH = "data/classes.xml"
 
-foes_by_mission: dict[str, list[Foe]] = {}
+from src.services.global_foes import foes_by_mission, link_foe_to_mission
+# foes_by_mission: dict[str, list[Foe]] = {}
```
Comment (Grimmys) [inline src/services/load_from_xml_manager.py:42]: Please remove this commented out line of code since it's not necessary anymore

**Review by Grimmys — CHANGES_REQUESTED** [review]
Sorry for the delay before my review, I didn't see the notification earlier.

Thanks for your contribution, it's nice changes 👍 

Except the few comments I put, I'm wondering if a better name could be found for `global_foes` service, I feel like this name doesn't say a lot. But it can be changed later on.

File: src/services/load_from_xml_manager.py (hunk 2 above)
Comment (silent-lindsay) [reply src/services/load_from_xml_manager.py]: Removed

**Review by Grimmys — APPROVED** [review]
